precompiles/avs/events.go

```diff
@@ -1,22 +1,328 @@
+		Data:        packed,
+		BlockNumber: uint64(ctx.BlockHeight()),
+	})
+
+	return nil
+}
+
+func (p Precompile) EmitOperatorJoined(ctx sdk.Context, stateDB vm.StateDB, params *avskeep.OperatorOptParams) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeOperatorJoined]
+
+	topics := make([]common.Hash, 1)
+
+	// The first topic is always the signature of the event.
+	topics[0] = event.ID
+
+	var err error
+
+	// Pack the arguments to be used as the Data field
+	arguments := event.Inputs[0:2]
+	packed, err := arguments.Pack(
+		common.HexToAddress(params.OperatorAddress),
+		true)
+	if err != nil {
+		return err
+	}
+
+	stateDB.AddLog(&ethtypes.Log{
+		Address:     p.Address(),
+		Topics:      topics,
+		Data:        packed,
+		BlockNumber: uint64(ctx.BlockHeight()),
+	})
+
+	return nil
+}
+
+func (p Precompile) EmitOperatorOuted(ctx sdk.Context, stateDB vm.StateDB, params *avskeep.OperatorOptParams) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeOperatorOuted]
+
+	topics := make([]common.Hash, 1)
+
+	// The first topic is always the signature of the event.
+	topics[0] = event.ID
+
+	var err error
+
+	// Pack the arguments to be used as the Data field
+	arguments := event.Inputs[0:2]
+	packed, err := arguments.Pack(
+		common.HexToAddress(params.OperatorAddress),
+		true)
+	if err != nil {
+		return err
+	}
+
+	stateDB.AddLog(&ethtypes.Log{
+		Address:     p.Address(),
+		Topics:      topics,
+		Data:        packed,
+		BlockNumber: uint64(ctx.BlockHeight()),
+	})
+
+	return nil
+}
+
+func (p Precompile) EmitTaskCreated(ctx sdk.Context, stateDB vm.StateDB, task *avskeep.TaskInfoParams) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeTaskCreated]
+
+	topics := make([]common.Hash, 1)
+
+	// The first topic is always the signature of the event.
+	topics[0] = event.ID
+
+	var err error
+
+	// Pack the arguments to be used as the Data field
+	arguments := event.Inputs[0:9]
+	packed, err := arguments.Pack(
+		common.HexToAddress(task.CallerAddress),
+		task.TaskID,
+		common.HexToAddress(task.TaskContractAddress),
+		task.TaskName,
+		task.Hash,
+		task.TaskResponsePeriod,
+		task.TaskChallengePeriod,
+		task.ThresholdPercentage,
+		task.TaskStatisticalPeriod)
+	if err != nil {
+		return err
+	}
+
+	stateDB.AddLog(&ethtypes.Log{
+		Address:     p.Address(),
+		Topics:      topics,
+		Data:        packed,
+		BlockNumber: uint64(ctx.BlockHeight()),
+	})
+
+	return nil
+}
+
+func (p Precompile) EmitChallengeInitiated(ctx sdk.Context, stateDB vm.StateDB, params *avskeep.ChallengeParams) error {
+	// Prepare the event topics
+	event := p.ABI.Events[EventTypeChallengeInitiated]
+
+	topics := make([]common.Hash, 1)
+
+	// The first topic is always the signature of the event.
+	topics[0] = event.ID
+
+	var err error
+
+	// Pack the arguments to be used as the Data field
+	arguments := event.Inputs[0:5]
+	packed, err := arguments.Pack(
+		common.HexToAddress(params.CallerAddress),
```

Comment:
Seemed like the arguments number didn't match with the `event.Inputs[0:5]` amount